### PR TITLE
unselected RNA-seq based workflow

### PR DESCRIPTION
**Pull request by mapo9**
This PR allows to run airrflow from unselected RNA-seq fastq files. This is done by using the TRUST4 tool.

* bulk: airrflow runs trust4 and uses **TRUST_airr.tsv** in the immcantation framework
* single-cell: airrflow runs trust4 and uses **TRUST_barcodes_airr.tsv** in the immcantation framework

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/airrflow/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/airrflow _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by ggabernet — COMMENTED** [review]
@mapo9 I've just started checking at the code and testing it. I already have some comments but still need to check it and test it further. Also I just merged the test data into the nf-core repo so the tests can run

File: subworkflows/local/rnaseq_input.nf [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        }
+        .set { ch_trust4_airr_file }
+
+
+    // create channel with airr file
+    ch_trust4_airr_file.bulk.mix ( ch_trust4_airr_file.sc ).set { ch_trust4_airr }
+
+
+    // rename tsv file to unique name
+    RENAME_FILE_TSV(
```
[the remark this answers is not in the document]
Comment (mapo9) [reply subworkflows/local/rnaseq_input.nf:106]: there was a bug at an earlier version of this PR w.r.t to renaming the fastp output. This was fixed though, maybe you need to pull the pr again?
If your error persists, please also feel free to send me the error so I can fix it.
This `RENAME_FILE_TSV` module is indeed not needed anymore with how things are handled now.
Thanks!

File: nextflow_schema.json [hunk 2]
```diff
@@ -123,7 +130,6 @@
             "properties": {
                 "umi_position": {
                     "type": "string",
-                    "default": "R1",
```
[the remark this answers is not in the document]
Comment (mapo9) [reply nextflow_schema.json]: I redid the changes and created a new trust4-specific parameter

File: nextflow_schema.json [hunk 3]
```diff
@@ -588,7 +626,6 @@
                 "config_profile_url": {
                     "type": "string",
                     "description": "Directory to keep pipeline Nextflow logs and reports.",
-                    "default": "${params.outdir}/pipeline_info",
```
Comment (ggabernet) [inline nextflow_schema.json:591]: oups I think this line should not be deleted

**Comment by mapo9** [comment]
Hi Gisela, 

I incorporated the changes we discussed. Among others, I updated the nf-core/trust4 module which now fits our use-case and switched to the nf-core module in airrflow as well.

File: subworkflows/local/fastq_input_check.nf [hunk 4]
```diff
@@ -84,12 +84,17 @@ def create_fastq_channels(LinkedHashMap col) {
...
+        if (params.umi_read == "") {
+            meta.umi_read = null
+        } else {
+            meta.umi_read = params.umi_read
+        }
+        array = [ meta, [ file(col.filename_R1), file(col.filename_R2) ] ]
     } else {
-
         array = [ meta, [ file(col.filename_R1), file(col.filename_R2) ] ]
-        if (params.index_file) {
```
Comment (ggabernet) [inline subworkflows/local/fastq_input_check.nf:90]: @mapo9 this part was the one causing the other processes to fail, I fixed it now

**Review by ggabernet — APPROVED** [review]
I think I fixed the issue that made tests fail, let's see if they pass now

**Pull request merged**